datasets/flwr_datasets/resplitter/divide_resplitter.py

```diff
@@ -0,0 +1,286 @@
+                    end_index += int(len(split_data) * size)
+                elif isinstance(size, int):
+                    end_index += size
+                else:
+                    raise ValueError(
+                        "The type of size value for the divide config must "
+                        "be int or float."
+                    )
+                if end_index > len(split_data):
+                    raise ValueError(
+                        "The size specified in the `divide_config` is greater than "
+                        "the size of the dataset."
+                    )
+                if end_index == start_index:
+                    raise ValueError(
+                        f"The size specified in the `divide_config` results in the "
+                        f"dataset of size 0. The problem occurred in {new_splits_dict}."
+                        f"Please make sure to provide sizes that do not produce empty"
+                        f"datasets."
+                    )
+                resplit_dataset[new_split_name] = split_data.select(
+                    range(start_index, end_index)
+                )
+                start_index = end_index
+        return datasets.DatasetDict(resplit_dataset)
+
+    def _check_duplicate_splits_in_config(self) -> None:
+        """Check if the new split names are duplicated in `divide_config`."""
+        if self._config_type == "single-split":
+            new_splits = list(self._single_split_config.keys())
+        elif self._config_type == "multiple-splits":
+            new_splits = []
+            for new_splits_dict in self._multiple_splits_config.values():
+                assert isinstance(new_splits_dict, dict)
+                new_values = list(new_splits_dict.keys())
+                assert isinstance(new_values, list)
+                new_splits.extend(new_values)
+        else:
+            raise ValueError("Incorrect type of config.")
+
+        duplicates = [
+            item for item, count in collections.Counter(new_splits).items() if count > 1
+        ]
+        if duplicates:
+            raise ValueError(
+                "The specified values of the new splits in "
+                "`divide_config` are duplicated. Please specify"
+                "unique values for each new split."
+            )
+
+    def _check_duplicate_splits_in_config_and_original_dataset(
+        self, dataset_splits: List[str]
+    ) -> None:
+        """Check duplicates along the new split values and dataset splits.
+
+        This check can happen only at the time this class is called (it does not have
+        access to the dataset prior to that).
+        """
+        if self._config_type == "single-split":
+            new_splits = list(self._single_split_config.keys())
+            all_splits = dataset_splits + new_splits
+            assert self._divide_split is not None
+            all_splits.pop(all_splits.index(self._divide_split))
```

Comment:
Why do we need this? in case the `_divide_split` split is going to be re-split but keeping the same name?

Reply:
In this case (single-split) the divide_split dictates which split will be divided. Here I try to collect all (new) splits -> the situation that will be after the split will be performed. So if I have {"train": train-data, "test": test-data} and I say to split e.g. train (self._divide_split = "train") into {"train": 0.5, "valid":0.5}, then all splits (`all_splits` ) will be ["train" (the new one), "valid", "test"]. and in the code I'm merging the original (`dataset_splits`) ["train", "test"] with the new (`new_splits`) ["train", "valid"]. What I need to do is to remove the original "train" from the sum of splits (the original train will be divided so that's why I do the pop on the `_divide_split` which is "train"